### PR TITLE
39253 SEO Title and description

### PR DESCRIPTION
**Pull request by dai-eastgate**
related to https://github.com/onOffice-Web-Org/oo-wp-plugin/issues/577

changed log :
Set limit character for title and description

### CODE REVIEW

File: plugin.php [hunk 1]
```diff
@@ -399,4 +403,24 @@ function update_status_close_action_button_option()
 	};
 }, 500);
 
+function limit_characters(string $text, int $max_length): string
```
Comment (andernath) [inline plugin.php:406]: Why is this function necessary? Wouldn't _substr($result['title'], 0, DEFAULT_LIMIT_CHARACTER_TITLE);_ do the same?

**Review by andernath — COMMENTED** [review]
@dai-eastgate I have a question in review. Please take a look.

File: plugin.php (hunk 1 above)
Comment (andernath) [reply plugin.php:406]: @dai-eastgate 
Oh I guess you do not want to cut a word in half? Like 15 chars for "Beautiful cottage" would be "Beautiful" and not "Beautiful cott". Right?

I will clarify if this is necessary with the team.

Comment (andernath) [reply plugin.php:406]: The Seo Team thinks it's a good idea not to cut off words at the end.

I tried to improve the Code a bit. What do you think @dai-eastgate ? (sorry for bad formatting) 

`$result = $pDI->get(EstateViewDocumentTitleBuilder::class)->buildDocumentTitle($title);`
`if (strlen($result['title']) > DEFAULT_LIMIT_CHARACTER_TITLE) {`
      `$shortenedTitle = substr($result['title'], 0, DEFAULT_LIMIT_CHARACTER_TITLE);`
  `if (substr($result['title'], DEFAULT_LIMIT_CHARACTER_TITLE, 1) != ' ') {`
      `$shortenedTitle = substr($shortenedTitle, 0, strrpos($shortenedTitle, ' '));`
   `}`
   `$result['title'] = $shortenedTitle;`
`}`
`return $result;`

Comment (yeneastgate) [reply plugin.php:406]: @andernath yes,I will try the way you suggested, and will report back the result, thanks

Comment (dai-eastgate) [reply plugin.php:406]: @andernath yes, we will try the way you suggested, and will report back the result, thanks

Comment (dai-eastgate) [reply plugin.php]: @andernath I tried your suggestion, and it worked well. Please review the code I have refactored. Thank you!

**Review by andernath — APPROVED** [review]
I will start testing

**Pull request merged**